src/freerct.cpp

```diff
@@ -108,6 +108,47 @@ static void PrintVersion()
 		"see <http://www.gnu.org/licenses/>\n");
 }
 
+/**
+ * Look for savegames and a config file in various locations we used before we implemented
+ * the XDG basedir specification, and move our findings to the new-style directories.
+ * Does not overwrite existing files.
+ * @todo Can be removed some time in the future (after v0.1) when nobody should have any unmigrated files anymore.
+ */
+static void MigrateOldFiles()
+{
+	const std::string userdata = freerct_userdata_prefix();
+	const std::string &homedir = GetUserHomeDirectory();
+
+	for (const std::string &old_directory : {homedir + "/.config/freerct", homedir + "/.local/share/freerct", homedir + "/.freerct"}) {
+		std::unique_ptr<DirectoryReader> dr(MakeDirectoryReader());
+		dr->OpenPath(old_directory.c_str());
+		const char *filename;
+		while ((filename = dr->NextEntry()) != nullptr) {
+			std::string name(filename);
+			name = name.substr(name.find_last_of(dr->dir_sep) + 1);
+			std::string destination;
+			if (name == "freerct.cfg") {
+				destination = userdata;
+				destination += dr->dir_sep;
+				destination += name;
+			} else if (name.size() > 4 && name.compare(name.size() - 4, 4, ".fct") == 0) {
```

Comment:
Here and in 130, check case-insensitively?

Not sure, but could some support functions for handling paths be useful, now it's all expanded where-ever something with paths is done.